crates/runtime/src/dataaccelerator/spice_sys/dataset_checkpoint/duckdb.rs

```diff
@@ -61,6 +61,10 @@ impl DatasetCheckpoint {
             .execute(&upsert, [&self.dataset_name])
             .map_err(|e| e.to_string())?;
 
+        duckdb_conn
```

Comment:
This will require a deeper fix to share the pool connection with the accelerator